python/edgar_filing_searcher/parsers/daily_index_crawler.py

```diff
@@ -0,0 +1,96 @@
+# pylint: disable=no-member
+"""This file crawls the daily index directory pages"""
+import logging
+import re
+import time
+from datetime import date, timedelta
+
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+
+from edgar_filing_searcher.errors import BadWebPageResponseException, InvalidUrlException
+
+
+def get_request_response(url):
+    """Returns the response request from the URL"""
+    backoff_factor_value = 3
+    timeout_value = 3
+
+    retry_strategy = Retry(
+        total=3,
+        status_forcelist=(403, 429, 500, 502, 503, 504),
+        allowed_methods=["GET"],
+        backoff_factor=backoff_factor_value,
+    )
+
+    session = requests.Session()
+
+    session.mount("http://", HTTPAdapter(max_retries=retry_strategy))
+    session.mount("https://", HTTPAdapter(max_retries=retry_strategy))
+
+    response = session.get(
+        url,
+        headers={"user-agent": "filing_13f_searcher"}, timeout=timeout_value
+    )
+    return response
+
+
+def get_text(url):
+    """Returns the html and text from the URL"""
+    response = get_request_response(url)
+    time.sleep(1)
+    full_text = response.text
+    logging.debug('Successfully ran get_text on URL %s', url)
+    return full_text
+
+
+def get_subdirectories_for_specific_date(full_date: date):
+    """Returns all cik number and accession number subdirectory strings for a specific date"""
+    base_url = "https://www.sec.gov/Archives/edgar/daily-index"
+    quarter = round((full_date.month / 4) + 1)
+    subdirectory_date_94 = full_date.strftime('%m%d%y')

```

Comment:
you could move each of these into the respective if block so that they don't all have to be calculated.